bufferobserver.go

```diff
@@ -0,0 +1,13 @@
+package main
+
+// BufferObserver separates observer functionality out of file.
+// A BufferObserver is something that can be kept track
+// of through the ObservableEditableBuffer.
+type BufferObserver interface {
+
+	// inserted is a callback function which updates the observer's texts.
```

Comment:
"inserted informs the implementer that rune array r was inserted at position q0"?